homeassistant/components/wemo/__init__.py

```diff
@@ -79,14 +79,21 @@ def coerce_host_port(value: str) -> HostPortTuple:
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up for WeMo devices."""
-    hass.data[DOMAIN] = {
-        "config": config.get(DOMAIN, {}),
+def _domain_data(config: ConfigType) -> ConfigType:
+    """Initialize the data for the wemo domain."""
+    return {
+        "config": config,
         "registry": None,
         "pending": {},
+        "platforms": set(),
+        "stop": None,
     }
```

Comment:
In a follow-up PR, let's make this a DataClass 👍

Reply:
Implemented in this PR based on feedback below.